### DISCUSSION TITLE
How do I actually debug Go and/or TS in LazyVim?

### DISCUSSION THREAD

**Discussion by banjo** [LazyVim/LazyVim#4948] [Q&A]
I currently have some problems to get the debugger working in LazyVim. I currently mostly use Go and TS and just want to be able to debug my applications. 

I am having a really hard time finding good guides/tutorials.

Im trying to start the debugging using using the provided keybinds but to no success. I assume there is some setup that needs to be done, either to run for example the `dev` script in package and debug, or to just debug a simple `go run main.go` or `tsx main.ts`.

Are there any good guides somewhere? I feel like the LazyVim site only cover the setup more or less.

**Comment by rogeriofrsouza** [comment]
Are you using the Extras?
They should have some debug keymaps in `<leader>d`

https://www.lazyvim.org/extras/lang/go
https://www.lazyvim.org/extras/lang/typescript

**Comment by azdanov** [comment]
I was confused a bit also, but figured out that I need to put a breakpoint first somewhere for the debugger to stop there.
